From e1c278ae825958186418cc8201a6f49f74b75181 Mon Sep 17 00:00:00 2001
From: Karol Herbst <git@karolherbst.de>
Date: Tue, 4 Jul 2023 02:15:03 +0200
Subject: [PATCH] clc: use CLANG_RESOURCE_DIR for clang's resource path

With certain build configuration that value can be a non empty string and
needs to be used.

This will also require distributions to rebuild mesa if and only if
CLANG_RESOURCE_DIR changes between clang rebuilds or updates.

Signed-off-by: Karol Herbst <git@karolherbst.de>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/23981>
---
 src/compiler/clc/clc_helpers.cpp | 4 +++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/compiler/clc/clc_helpers.cpp b/src/compiler/clc/clc_helpers.cpp
index 412a0dbf93de3..0daf06f1e7cd8 100644
--- a/src/compiler/clc/clc_helpers.cpp
+++ b/src/compiler/clc/clc_helpers.cpp
@@ -39,6 +39,7 @@
 #include <llvm-c/Target.h>
 #include <LLVMSPIRVLib/LLVMSPIRVLib.h>
 
+#include <clang/Config/config.h>
 #include <clang/Driver/Driver.h>
 #include <clang/CodeGen/CodeGenAction.h>
 #include <clang/Lex/PreprocessorOptions.h>
@@ -76,6 +77,7 @@ using ::llvm::Function;
 using ::llvm::LLVMContext;
 using ::llvm::Module;
 using ::llvm::raw_string_ostream;
+using ::clang::driver::Driver;
 
 static void
 llvm_log_handler(const ::llvm::DiagnosticInfo &di, void *data) {
@@ -877,7 +879,7 @@ clc_compile_to_llvm_module(LLVMContext &llvm_ctx,
    // because we might have linked clang statically.
    auto libclang_path = fs::path(LLVM_LIB_DIR) / "libclang.so";
    auto clang_res_path =
-      fs::path(clang::driver::Driver::GetResourcesPath(libclang_path.string())) / "include";
+      fs::path(Driver::GetResourcesPath(libclang_path.string(), CLANG_RESOURCE_DIR)) / "include";
 
    c->getHeaderSearchOpts().UseBuiltinIncludes = true;
    c->getHeaderSearchOpts().UseStandardSystemIncludes = true;
-- 
GitLab

